taar/recommenders/similarity_recommender.py

```diff
@@ -0,0 +1,155 @@
+        REQUIRED_FIELDS = ["geoCity", "subsessionLength", "locale", "os", "bookmark_count", "tab_open_count",
+                           "total_uri", "unique_tlds"]
+
+        has_fields = all([client_data.get(f, None) is not None for f in REQUIRED_FIELDS])
+        if not has_fields:
+            # Can not add extra info because client_id may not be available.
+            logger.error("Unusable client data encountered")
+        return has_fields
+
+    def get_lr(self, score):
+        """Compute a :float: likelihood ratio from a provided similarity score when compared
+        to two probability density functions which are computed and pre-loaded during init.
+
+        The numerator indicates the probability density that a particular similarity score
+        corresponds to a 'good' addon donor, i.e. a client that is similar in the sense of
+        telemetry variables. The denominator indicates the probability density that a particular
+        similarity score corresponds to a 'poor' addon donor
+
+        :param score: A similarity score between a pair of objects.
+        :returns: The approximate float likelihood ratio corresponding to provided score.
+        """
+        index = [abs(score - s[0]) for s in self.lr_curves]
+
+        # Find the index of the closest value that was precomputed in lr_curves
+        _, idx = min((val, idx) for (idx, val) in enumerate(index))
+
+        numer_val = self.lr_curves[idx][1][0]
+        denum_val = self.lr_curves[idx][1][1]
+
+        # Compute LR based on numerator and denominator values
+        return float(numer_val) / float(denum_val)
+
+    def get_similar_donors(self, client_data):
+        """Computes a set of :float: similarity scores between a client and a set of candidate
+        donors for which comparable variables have been measured.
+
+        A custom similarity metric is defined in this function that combines the Hamming distance
+        for categorical variables with the Canberra distance for continuous variables into a
+        univariate similarity metric between the client and a set of candidate donors loaded during
+        init.
+
+        :param client_data: a client data payload including a subset fo telemetry fields.
+        :rtype: :list:`tuples`: the approximate likelihood ratio corresponding to the
+        internally computed similarity score and a list of addons taken from that donor
+        """
+        client_categorical_feats = [client_data.get(specified_key) for specified_key in CATEGORICAL_FEATURES]
+        client_continuous_feats = [client_data.get(specified_key) for specified_key in CONTINUOUS_FEATURES]
+
+        donor_set = []
+        donor_index = 0
+
+        # This could be optimized by using cdist, however both logging and debugging benefit from having it in a loop.
+        # If performance improvements are necessary in he future, this is a good place to start.
+        for donor in self.donors_list:
```

Comment:
Instead of manually tracking the `donor_index`, let's use `enumerate` here. This line then becomes: `for donor_index, donor in enumerate(self.donors_list):`. Let's also remove the variable declaration on line 107.